zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/ClusterConfigurationManagementRequestsHandler.java

```diff
@@ -148,31 +148,22 @@ public ActorFuture<ClusterConfiguration> getTopology() {
 
   private ActorFuture<ClusterConfigurationChangeResponse> handleRequest(
       final boolean dryRun, final ConfigurationChangeRequest request) {
-    final ActorFuture<ClusterConfigurationChangeResponse> responseFuture = executor.createFuture();
     final Function<ConfigurationChangeRequest, ActorFuture<ConfigurationChangeResult>> handler;
     if (dryRun) {
       handler = coordinator::simulateOperations;
     } else {
       handler = coordinator::applyOperations;
     }
-    executor.run(
-        () ->
-            handler
-                .apply(request)
-                .onComplete(
-                    (result, error) -> {
-                      if (error == null) {
-                        final var changeStatus =
-                            new ClusterConfigurationChangeResponse(
-                                result.changeId(),
-                                result.currentConfiguration().members(),
-                                result.finalConfiguration().members(),
-                                result.operations());
-                        responseFuture.complete(changeStatus);
-                      } else {
-                        responseFuture.completeExceptionally(error);
-                      }
-                    }));
-    return responseFuture;
+
+    return handler
```

Comment:
:thought_balloon: There's many places in this class where we do `executor.run(() -> ...)`, but then the first thing we do in the callback is call something which again calls `executor.run()` :thinking: 

It seems to me we could keep the semantics here the same by just dropping the initial `executor.run` anyway.